Source/UI/Home/HomeView.swift

```diff
@@ -0,0 +1,314 @@
+        }
+        .environmentObject(botRepository)
+        .navigationTitle(Localized.home.text)
+        .background(Color.appBg)
+        .toolbar {
+            ToolbarItemGroup(placement: .navigationBarTrailing) {
+                Button {
+                    helpDrawerState.isShowingHomeHelpDrawer = true
+                } label: {
+                    Image.navIconHelp
+                }
+                .popover(isPresented: $helpDrawerState.isShowingHomeHelpDrawer) {
+                    Group {
+                        if #available(iOS 16.0, *) {
+                            HelpDrawerCoordinator.helpDrawerView(for: helpDrawer) {
+                                helpDrawerState.isShowingHomeHelpDrawer = false
+                            }
+                            .presentationDetents([.medium])
+                        } else {
+                            HelpDrawerCoordinator.helpDrawerView(for: helpDrawer) {
+                                helpDrawerState.isShowingHomeHelpDrawer = false
+                            }
+                        }
+                    }
+                    .onAppear {
+                        Analytics.shared.trackDidShowScreen(screenName: helpDrawer.screenName)
+                        HelpDrawerCoordinator.didShowHelp(for: helpDrawer)
+                    }
+                }
+                Button {
+                    Analytics.shared.trackDidTapButton(buttonName: "compose")
+                    showCompose()
+                } label: {
+                    Image.navIconWrite
+                }
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didChangeHomeFeedAlgorithm)) { _ in
+            Task.detached {
+                await loadFromScratch()
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didUpdateRelationship)) { _ in
+            Task.detached {
+                await loadFromScratch()
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didPublishPost)) { _ in
+            Task.detached {
+                await loadFromScratch()
+            }
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .didRefresh)) { _ in
+            Task.detached {
+                await checkNewItemsIfNeeded()
+            }
+        }
+        .onAppear {
+            CrashReporting.shared.record("Did Show Home")
+            Analytics.shared.trackDidShowScreen(screenName: "home")
+            HelpDrawerCoordinator.showFirstTimeHelp(for: helpDrawer, state: helpDrawerState)
+        }
+    }
+
+    private func showCompose() {
+        let controller = NewPostViewController()
+        controller.didPublish = { post in
+            NotificationCenter.default.post(.didPublishPost(post))
+        }
+        let navController = UINavigationController(rootViewController: controller)
+        AppController.shared.present(navController, animated: true)
```

Comment:
Probably `AppController` should be an `@Environment` var as well? I don't think it matters much at the moment, but if we ever want to support multiple windows/scenes on macOS/iPadOS then it's a good idea to lay the groundwork now by keeping singletons out of our SwiftUI views.

Reply:
👌